lib/src/build_context_impl.dart

```diff
@@ -39,12 +39,12 @@ extension MediaQueryExt on BuildContext {
 }
 
 extension NavigatorExt on BuildContext {
-  Future<T> push<T>(Route<T> route) => Navigator.push(this, route);
+  Future<T?> push<T>(Route<T> route) => Navigator.push(this, route);
 
-  void pop<T extends Object>([T result]) => Navigator.pop(this, result);
+  void pop<T extends Object>([T? result]) => Navigator.pop(this, result);
 
-  Future<Object> pushNamed<T>(String routeName, {Object arguments}) =>
-      Navigator.pushNamed(this, routeName, arguments: arguments);
+  Future<Object?> pushNamed<T>(String routeName, {Object? arguments}) =>
```

Comment:
@pedromassango Why is the return type `Object?`? Shouldn't this be `T?`?

Reply:
Because `Navigator.pushNamed` expect objects:

`static Future<T?> pushNamed<T extends Object?>`

I think you can change it now :)

Reply:
I think we can do the same with the `arguments` type

Reply:
I don't think we should do this, because the type of `Navigator.pushNamed` is also `Object` for `arguments`:

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/navigator.dart#L1721